Modules/Devices/src/Devices.Domain/Entities/Identities/Identity.cs

```diff
@@ -100,6 +103,22 @@ public void DeletionProcessApprovalReminder3Sent()
         deletionProcess.ApprovalReminder3Sent(Address);
     }
 
+    public IdentityDeletionProcess ApproveDeletionProcess(DeviceId deviceId, string deletionProcessId)
+    {
+        var deletionProcess = DeletionProcesses.FirstOrDefault(x => x.Id == deletionProcessId) ?? throw new DomainException(GenericDomainErrors.NotFound(nameof(IdentityDeletionProcess)));
+
+        if (deletionProcess.Status != DeletionProcessStatus.WaitingForApproval)
+            throw new DomainException(DomainErrors.NoDeletionProcessFoundInCorrectStatusForApproval());
+
+        deletionProcess.Approve(Address, deviceId);
+
+        Status = IdentityStatus.ToBeDeleted;
+        DeletionGracePeriodEndsAt = deletionProcess.GracePeriodEndsAt;
+        TierId = Tier.QUEUED_FOR_DELETION.Id;
```

Comment:
Before we change the tier, we have to save the current tier in a "TierBeforeDeletion" property, so that we can reset the Tier in case the deletion process is canceled